test_embark/locking.js

```diff
@@ -0,0 +1,405 @@
+    // lock again
+    await staking.lock(defaultAmount / 4, user1, web3.utils.asciiToHex('')).send()
+
+    // unlock
+    await staking.unlockAll(owner).send({ from: user1 })
+
+    assert.equal((await staking.unlockedBalanceOf(owner).call()).valueOf(), defaultAmount, "Unlocked balance should match")
+    assert.equal((await staking.locksCount(owner).call()).valueOf(), 0, "there shouldn't be locks")
+  })
+
+  it('unlocks all with no previous locks', async () => {
+    await staking.unlockAll(owner).send({ from: user1 })
+    assert.equal((await staking.locksCount(owner).call()).valueOf(), 0, "there shouldn't be locks")
+  })
+
+  it('tries to unlockAll but it only unlocks one', async () => {
+    const lockId = await approveStakeAndLock(user2)
+    // lock again, different EOA manager
+    await staking.lock(defaultAmount / 4, user1, web3.utils.asciiToHex('')).send()
+
+    // unlock
+    await staking.unlockAll(owner).send({ from: user1 })
+
+    assert.equal((await staking.unlockedBalanceOf(owner).call()).valueOf(), defaultAmount / 2, "Unlocked balance should match")
+    assert.equal((await staking.locksCount(owner).call()).valueOf(), 1, "there shouldn't be locks")
+  })
+
+  it('fails trying to unlockAllOrNone if a lock cannot be unlocked', async () => {
+    const lockId = await approveStakeAndLock(user1)
+    // lock again, different EOA manager
+    await staking.lock(defaultAmount / 4, user2, web3.utils.asciiToHex('')).send()
+
+    // unlock
+    return assertRevert(async () => {
+      await staking.unlockAllOrNone(owner).send({ from: user1 })
+    })
+  })
+
+  it('change lock amount', async () => {
+    const lockId = await approveStakeAndLock(lockManagerAddress)
+    const lock = await staking.getLock(owner, lockId).call()
+    assert.equal(lock[0], defaultAmount / 2, "Amount should match")
+    assert.equal((await staking.unlockedBalanceOf(owner).call()).valueOf(), defaultAmount / 2, "Unlocked balance should match")
+
+    // change amount
+    await lockManager.setLockAmount(stakingAddress, owner, lockId, defaultAmount / 4).send()
+
+    const lockAfter = await staking.getLock(owner, lockId).call()
+    assert.equal(lockAfter[0], defaultAmount / 4, "Amount should match")
+    assert.equal((await staking.unlockedBalanceOf(owner).call()).valueOf(), 3 * defaultAmount / 4, "Unlocked balance should match")
+  })
+
+  it('fails to change lock amount to zero', async () => {
+    const lockId = await approveStakeAndLock(lockManagerAddress)
+
+    // try to change amount
+    return assertRevert(async () => {
+      await lockManager.setLockAmount(stakingAddress, owner, lockId, 0).send()
+    })
+  })
+
+  it('fails to change lock amount to greater than before', async () => {
+    const lockId = await approveStakeAndLock(lockManagerAddress)
+
+    // try to change amount
+    return assertRevert(async () => {
+      await lockManager.setLockAmount(stakingAddress, owner, lockId, defaultAmount / 2 + 1).send()
```

Comment:
This feels like a magic number that would be solved by having a defaultLockAmount constant